users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -271,11 +271,223 @@ def run_scf_baseline():
+                    "deacy_epochs": "24_10000_5000",
+                },
+            ),
+        },
+        num_epochs=426,
+        evaluation_epochs=[374, 400, 426],
+        prefix="conformer_bs5k_decay_",
+    )
+
+    returnn_root = CloneGitRepositoryJob(
+        "https://github.com/rwth-i6/returnn",
+        commit="c4d36d06f6465e82a50d400d114259e07b8b0709",
+    ).out_repository
+    returnn_root.hash_overwrite = "returnn_conv_padding"
+    report = run_nn_args(
+        nn_args,
+        report_args_collection,
+        dev_corpora,
+        "report_scf_baseline_bs_decay.csv",
+        returnn_root=returnn_root,
+        recog_args={"epochs": [374, 400, 426, "best"]},
+    )
+    return report
+
+
+def run_scf_frozen_features():
+    gs.ALIAS_AND_OUTPUT_SUBDIR = "experiments/switchboard/ctc/feat/"
+
+    _, nn_system = run_scf_baseline()
+
+    (
+        returnn_datasets,
+        rasr_loss_corpus_path,
+        rasr_loss_corpus_segments,
+        rasr_loss_lexicon_path,
+        dev_corpora,
+    ) = get_datasets()
+
+    # Common configurations
+    common_returnn_args = {
+        "batch_size": 5000,
+        "rasr_binary_path": RASR_BINARY_PATH,
+        "rasr_loss_corpus_path": rasr_loss_corpus_path,
+        "rasr_loss_corpus_segments": rasr_loss_corpus_segments,
+        "rasr_loss_lexicon_path": rasr_loss_lexicon_path,
+        "datasets": returnn_datasets,
+        "conformer_type": "wei",
+        "specaug_old": {"max_feature": 15},
+    }
+
+    common_feature_args = {
+        "class": "ScfNetwork",
+        "size_tf": 256 // 2,
+        "stride_tf": 10 // 2,
+        "preemphasis": 0.97,
+        "wave_norm": True,
+    }
+
+    common_lr_args = {
+        "peak_lr": 4e-4,
+        "start_lr": 1.325e-05,
+        "end_lr": 1e-5,
+        "increase_epochs": 180,
+        "decrease_epochs": 180,
+        "final_epochs": 0,
+    }
+
+    common_report_args = {
+        "architecture": "conf-wei",
+        "lr": "wei_peak_4e-4_e450_cycle360",
+        "specaug": "wei_adapt_80dim",
+        "wave_norm": "True",
+        "preemphasis": "0.97",
+    }
+
+    nn_args, report_args_collection = get_nn_args_baseline(
+        nn_base_args={
+            "scf_baseline": {
+                "returnn_args": {
+                    "staged_opts": {
+                        1: "freeze_features",
+                        16: "unfreeze_features",
+                    },
+                    "extra_args": {
+                        "accum_grad_multiple_step": 2,
+                        "conv_pad_seq_len_to_power": 1.5,
+                        "preload_from_files": {
+                            "existing-model": {
+                                "filename": nn_system.train_jobs[
+                                    "conformer_bs2x5k_scf_baseline_preemphasis97_wn"
+                                ].out_checkpoints[400],
+                                "init_for_train": True,
+                                "prefix": "features",
```

Comment:
I removed the experiment since it did not make sense it it current form anyways

Reply:
Can you explain? What does not make sense? And why do you exclude it instead of fixing the problem?

Reply:
The experiment used features from a finished training and trained them again. This did not improve the result the last time I tried it and I would change it so that the already trained feature layer stays frozen for the entire time.

Reply:
For the mapping, I am not sure if it is needed here. I am currently testing this

Reply:
It is needed. I added it and changed the experiment to keep the features frozen.